### PR TITLE
Support parsing Oracle SHOW ERRORS; sql

### LINKED ISSUE #27167: Support parsing Oracle SHOW ERRORS; sql
# Background
Hi community,
This issue is for #26878.

ShardingSphere parser engine helps users parse a SQL to get the AST (Abstract Syntax Tree) and visit this tree to get SQLStatement (Java Object). Currently, we are planning to enhance the support for Oracle SQL parsing in ShardingSphere.

More details:
https://shardingsphere.apache.org/document/current/en/reference/sharding/parse/

# Task
This issue is to support more oracle sql parse, as follows:
```sql
SHOW ERRORS;
```

```sql
SHOW RETENTION POLICY;
SHOW DEFAULT DEVICE TYPE;
```

```sql
SHOW PARAMETER SMTP_OUT_SERVER
```

```sql
SPOOL onlineall.sql
SELECT 'ALTER DATABASE DATAFILE '''||name||''' ONLINE;' FROM V$DATAFILE;
```

```sql
SWITCH DATAFILE 4 TO COPY;
```

# Process
1. First confirm that this is a correct oracle sql syntax, if not please ignore;
2. Compare SQL definitions in Oficial SQL Doc and ShardingSphere SQL Doc;
3. If there is any difference in ShardingSphere SQL Doc, please correct them by referring to the Official SQL Doc;
4. Run mvn install the current_file_module;
5. Check whether there are any exceptions. If indeed, please fix them. (Especially xxxVisitor.class);
6. Add new corresponding SQL case in SQL Cases and expected parsed result in Expected Statment XML;
7. Run SQLParserParameterizedTest to make sure no exceptions.

# Relevant Skills
1. Master JAVA language
2. Have a basic understanding of Antlr `g4` file
3. Be familiar with Oracle SQLs

### PR DESCRIPTION
**Pull request by YaoFly**
Fixes #27167.

Changes proposed in this pull request:
  -

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by strongduanmu** [comment]
Hi @YaoFly, can you execute `./mvnw spotless:check -Pcheck -T1C` on your local machine?

**Comment by strongduanmu** [comment]
Hi @YaoFly, can you comment on the issue? I will assign it to you.

**Comment by YaoFly** [comment]
> Hi @YaoFly, can you comment on the issue? I will assign it to you.

ok.

File: parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DALStatement.g4 [hunk 1]
```diff
@@ -30,3 +30,32 @@ dbLinkInfo
...
+    : DOT_ identifier
+    ;
+
+spoolFileName
+    : identifier fileExt?
+    ;
+
+spool
+    : SPOOL spoolFileName
+    ;
```
Comment (strongduanmu) [inline parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DALStatement.g4:61]: Please add new line at the end of file.

File: parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/OracleKeyword.g4 [hunk 2]
```diff
@@ -7799,3 +7799,15 @@ NONBLOCKING
...
+    : D E V I C E
+    ;
+
+PARAMETER
+    :P A R A M E T E R
+    ;
+
+SPOOL
+    :S P O O L
+    ;
```
Comment (strongduanmu) [inline parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/OracleKeyword.g4:7813]: Please add new line at the end of file.

File: parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DALStatement.g4 [hunk 3]
```diff
@@ -30,3 +30,32 @@ dbLinkInfo
 explain
     : EXPLAIN PLAN (SET STATEMENT_ID EQ_ stringLiterals)? (INTO (schemaName DOT_)? tableName (AT_ dbLinkInfo)? )? FOR (insert | delete | update | select)
     ;
+
+parameterName
+    : identifier
+    ;
+
+showErrors
```
Comment (strongduanmu) [inline parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DALStatement.g4:38]: According to oracle doc - https://docs.oracle.com/en/database/oracle/oracle-database/19/sqpug/SHOW.html#GUID-6BB1499D-E537-43D1-A209-401F5DB95E16, It seems that these are all part of the show statement, except that it has different options.

File: parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DALStatement.g4 [hunk 4]
```diff
@@ -30,3 +30,32 @@ dbLinkInfo
...
+show
+    : showErrors
+    | showParameter
+    ;
+
+fileExt
+    : DOT_ identifier
+    ;
+
+spoolFileName
```
Comment (strongduanmu) [inline parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DALStatement.g4:55]: Can you support spool statement in next pr?

File: test/it/parser/src/main/resources/case/dml/select.xml [hunk 5]
```diff
@@ -9217,4 +9217,34 @@
             </expr>
         </where>
     </select>
+
+    <select sql-case-id="select_views">
```
Comment (strongduanmu) [inline test/it/parser/src/main/resources/case/dml/select.xml:9221]: Please remove this irrelevant case, this issue is focused on supporting the SHOW statement.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DALStatement.g4 (hunk 3 above)
Comment (YaoFly) [reply parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DALStatement.g4:38]: ok, I will complete all the options for show commend

File: parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DALStatement.g4 (hunk 4 above)
Comment (YaoFly) [reply parser/sql/dialect/oracle/src/main/antlr4/imports/oracle/DALStatement.g4:55]: yes i can.

**Comment by strongduanmu** [comment]
Hi @YaoFly, can you solve code conflict, and modify this pr according to review comment?

**Comment by YaoFly** [comment]
> Hi @YaoFly, can you solve code conflict, and modify this pr according to review comment?

yes. I will commit it today.

**Comment by YaoFly** [comment]
@strongduanmu Parsing of sqlCaseId as 'pl_sql_create_procedure' fails because the SHOW SQLCODE sql new the SQLCODE keyword. it is contains the following sql.
``DBMS_OUTPUT.PUT_LINE(SQLCODE || '---' || SQLERRM);
Support for DBMS_OUTPUT SQL seems to be needed to resolve this problem. need i support DBMS_OUTPUT statement in next pr?

File: parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/type/OracleDALStatementVisitor.java [hunk 6]
```diff
@@ -56,4 +58,10 @@ public ASTNode visitExplain(final ExplainContext ctx) {
         result.getVariableNames().addAll(getVariableNames());
         return result;
     }
+    
+    @Override
+    public ASTNode visitShow(final ShowContext ctx) {
+        return new OracleShowStatement();
+    }
+
```
Comment (strongduanmu) [inline parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/type/OracleDALStatementVisitor.java:66]: Please remove this uesless blank line.

File: test/it/parser/src/main/resources/case/dml/select.xml [hunk 7]
```diff
@@ -9359,4 +9359,5 @@
             </expr>
         </where>
     </select>
+
```
Comment (strongduanmu) [inline test/it/parser/src/main/resources/case/dml/select.xml:9362]: Please remove this uesless blank line.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/oracle/dal/OracleShowStatement.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+package org.apache.shardingsphere.sql.parser.sql.dialect.statement.oracle.dal;
+
+import org.apache.shardingsphere.sql.parser.sql.common.statement.AbstractSQLStatement;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.dal.DALStatement;
+import org.apache.shardingsphere.sql.parser.sql.dialect.statement.oracle.OracleStatement;
+
+/**
+ * Oracle show errors statement.
```
Comment (strongduanmu) [inline parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/oracle/dal/OracleShowStatement.java:25]: Please modify this javadoc.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Review by strongduanmu — APPROVED** [review]
Looks great, thank you for your contribution.

**Pull request merged**